src/compiler/checker.ts

```diff
@@ -16871,23 +16848,60 @@ namespace ts {
             return isLengthPushOrUnshift || isElementAssignment;
         }
 
-        function maybeTypePredicateCall(node: CallExpression) {
-            const links = getNodeLinks(node);
-            if (links.maybeTypePredicate === undefined) {
-                links.maybeTypePredicate = getMaybeTypePredicate(node);
+        function isDeclarationWithExplicitTypeAnnotation(declaration: Declaration | undefined) {
+            return !!(declaration && (

```

Comment:
This should handle `jsdoc` declarations, which means checking for the jsdoc node kinds and using `getEffectiveTypeAnnotation`.

Reply:
Yeah, I suppose.